### PR TITLE
Enhanced the module unload test

### PR DESCRIPTION
**Pull request by narasimhan-v**
Enhanced the test to run for all loaded modules.
Fixed the module test to skip the modules which have
dependencies.
Also, changed the way exit status is given, to check for errors
first.

Enhanced as per suggestion by @sacsant 

Signed-off-by: Narasimhan V <sim@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by narasimhan-v** [comment]
@PraveenPenguin Since this test would run for a huge number of tests based on the latest code, I hope we can consider output snippet from @sacsant in #243 for output verification

**Comment by narasimhan-v** [comment]
@sacsant The test runs for a huge number of drivers, for 100 counts. Is this ok ?

```
# find /lib/modules/$(uname -r)/ -name \*.ko | awk -F"/" '{print $NF}'|sed 's/\.ko//g' | wc -l
4501
```

**Comment by sacsant** [comment]
Yes 100 count will take a long time. I was thinking of 5 or 10. If we can make the count as a input it will  help.

**Comment by narasimhan-v** [comment]
@sacsant @PraveenPenguin have updated this commit with above suggestion

**Comment by PraveenPenguin** [comment]
@narasimhan-v sure need to give a try

File: io/driver/module_unload_load.sh [hunk 1]
```diff
@@ -13,13 +13,15 @@
 #
 # Copyright: 2016 IBM
 # Author: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>
+# Copyright: 2017 IBM
+# Authors: Narasimhan V <sim@linux.vnet.ibm.com>
```
Comment (PraveenPenguin) [inline io/driver/module_unload_load.sh]: I wouldn't call this a big change, would you? (the purpose of authorship is to mark significant changes )

Comment (PraveenPenguin) [inline io/driver/module_unload_load.sh]: example when the module is completely reworked or extended in a such extend that the original work is not that significant anymore.

**Comment by PraveenPenguin** [comment]
@narasimhan-v job log please as i don't have any system right now where i can try

**Comment by narasimhan-v** [comment]
@PraveenPenguin Sure, will provide later today

**Comment by narasimhan-v** [comment]
[job_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/766946/job_log.txt)

**Comment by PraveenPenguin** [comment]
@narasimhan-v i think may you be you  attached wrong job log as this job Interrupted by ctrl+c can you please have a look.?

```
2017-02-10 09:22:10,187 test             L0495 DEBUG|  -> self <class 'avocado.core.test.SimpleTest'>: 1-module_unload_load.sh
2017-02-10 09:22:10,188 test             L0625 ERROR| ERROR 1-module_unload_load.sh -> TestError: KeyboardInterrupt(): 
2017-02-10 09:22:10,188 test             L0612 INFO | 
2017-02-10 09:22:10,195 runner           L0054 ERROR| Runner error occurred: Interrupted by ctrl+c
Original status: ERROR
```

Regards
Praveen

**Comment by narasimhan-v** [comment]
@PraveenPenguin The test runs for around 4500 modules. So, I interrupted it.. Would you need job log for complete run ?

**Comment by PraveenPenguin** [comment]
@narasimhan-v never mind i already tried in my System and seem me working fine

**Comment by narasimhan-v** [comment]
@PraveenPenguin Thanks a lot

**Pull request merged**